### DISCUSSION TITLE
Integrating online calendars with Day Planner using the ICS plugin

### DISCUSSION THREAD

**Discussion by FeralFlora** [ivan-lednev/obsidian-day-planner#298] [Show and tell]
It was fairly easy to integrate my Google calendars into my Day planner workflow using the [ICS plugin.](https://github.com/muness/obsidian-ics)

I made this Templater/Javascript code, adapted from the example in the ICS repo, which plays well with Day planner:

```javascript
<%*
var events = await app.plugins.getPlugin('ics').getEvents(moment(tp.file.title,'YYYY-MM-DD'));
events.sort((a,b) => a.utime - b.utime).forEach((e) => {
    const calendar = e.icsName.replace(" ","-");
    const callUrl = e.callUrl? ` [${e.callType}](${e.callUrl})` : '';
    const location = e.location? ` 📍 *(${e.location})*` : '';
    tR+=`- [ ] ${e.time}-${e.endTime} **${e.summary}**${location}${callUrl} #${calendar}\n`
    if (e.description) {
        const inlineDescription = e.description.replace(/(?:\r\n|\r|\n)/g," ").replace(/\s{2,}/g," ");
        tR+=`    - ${inlineDescription}\n`
    }
})
%>
```

These are the results 👇 

## Screenshots

![278302580-20ca24cf-da7b-4271-b981-0c3ccfbb15ad](https://github.com/ivan-lednev/obsidian-day-planner/assets/33608083/0aad7fa8-a587-44e9-b0f0-ac0fcd6b934c)

![278302445-5b59c674-0714-4a6f-9078-d8760c03336a](https://github.com/ivan-lednev/obsidian-day-planner/assets/33608083/8ddc91ef-17f9-4fbe-9656-3c86321579fc)

https://github.com/muness/obsidian-ics/discussions/74

**Comment by TheFern2** [comment]
@FeralFlora Can you please a bit more about your setup. I tried setting up a template but couldn't get it to work. Getting date errors in the console when inserting the template.

Edit: Ah nvm it works fine when inserting template in the daily note. Duh user error.

**Comment by ivan-lednev** [comment]
From 0.19.0 the plugin has built-in support for online calendars.
